cmd/kubeadm/app/phases/certs/pkiutil/pki_helpers.go

```diff
@@ -200,6 +200,38 @@ func TryLoadKeyFromDisk(pkiPath, name string) (*rsa.PrivateKey, error) {
 	return key, nil
 }
 
+// TryLoadPrivatePublicKeyFromDisk tries to load the key from the disk and validates that it is valid
+func TryLoadPrivatePublicKeyFromDisk(pkiPath, name string) (*rsa.PrivateKey, *rsa.PublicKey, error) {
+	privateKeyPath := pathForKey(pkiPath, name)
+
+	// Parse the private key from a file
+	privKey, err := certutil.PrivateKeyFromFile(privateKeyPath)
+	if err != nil {
+		return nil, nil, fmt.Errorf("couldn't load the private key file %s: %v", privateKeyPath, err)
+	}
+
+	publicKeyPath := pathForPublicKey(pkiPath, name)
+
+	// Parse the public key from a file
+	pubKeys, err := certutil.PublicKeysFromFile(publicKeyPath)
+	if err != nil {
+		return nil, nil, fmt.Errorf("couldn't load the public key file %s: %v", publicKeyPath, err)
+	}
+
+	var key *rsa.PrivateKey
+	// Allow RSA format only
+	if k, ok := privKey.(*rsa.PrivateKey); ok {
```

Comment:
nit on style, this can be moved out of the if clause and you can check for `!ok`, return in that case, otherwise just use `:=` afterwards.

Generally I'm trying to minimize the amount of indentation in funcs and avoid if/else when possible. This is still a nit though ;)

Reply:
Fixed